File: recycleview3.0/app/src/main/java/com/example/recycleview30/FruitAdapter.java
package com.example.recycleview30;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class FruitAdapter extends RecyclerView.Adapter<FruitAdapter.viewholder> {

    public List<Fruit> fruitList;

    static class viewholder extends RecyclerView.ViewHolder {

        ImageView fruitimage;

        TextView fruitname;

        public viewholder(@NonNull View itemView) {
            super(itemView);
            fruitimage = (ImageView) itemView.findViewById(R.id.fruit_image);
            fruitname = (TextView) itemView.findViewById(R.id.fruit_name);
        }
    }

    public FruitAdapter(List<Fruit> fruitList) {
        this.fruitList = fruitList;
    }


    @NonNull
    @Override
    public viewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Log.d("123", "onCreateViewHolder: creat" + viewType);
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.fruit_item,parent,false);
        return new viewholder(view);

    }

    @Override
    public void onBindViewHolder(@NonNull viewholder holder, int position) {
        Log.d("456", "onBindViewHolder: bind" + position);
        Fruit fruit = fruitList.get(position);
        holder.fruitname.setText(fruit.getName());
        holder.fruitimage.setImageResource(fruit.getImageid());
    }


    @Override
    public int getItemCount() {
        return fruitList.size();
    }
}

File: RecycleViewPractice_review/app/src/main/java/com/example/recycleviewpractice_review/FruitAdapter.kt
package com.example.recycleviewpractice_review

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class FruitAdapter(var fruitList: ArrayList<Fruit>):
    RecyclerView.Adapter<FruitAdapter.ViewHolder>() {
    inner class ViewHolder(view: View): RecyclerView.ViewHolder(view) {
        val fruitName: TextView = view.findViewById(R.id.fruit_name)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.fruit_item, parent, false);
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return fruitList.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val fruit = fruitList[position]
        holder.fruitName.text = fruit.name
    }

    fun removeData(position: Int){
        fruitList.removeAt(position)
        notifyItemRemoved(position)
    }
}